packages/expo-modules-core/ios/JSI/ExpoModulesHostObject.mm

```diff
@@ -0,0 +1,39 @@
+// Copyright 2022-present 650 Industries. All rights reserved.
+
+#import <ExpoModulesCore/ExpoModulesHostObject.h>
+
+namespace expo {
+
+ExpoModulesHostObject::ExpoModulesHostObject(SwiftInteropBridge *swiftInterop) : swiftInterop(swiftInterop) {}
+
+ExpoModulesHostObject::~ExpoModulesHostObject() {
+  [swiftInterop setRuntime:nil];
```

Comment:
is it intentionally to `setRuntime:nil` here? what if someone do `global.ExpoModules = null;` in javascript and after GC, will it also call to here and cleanup the runtime?

Reply:
Yeah, it's intentional. Our `JavaScriptRuntime` wrapper holds the reference to `jsi::Runtime` and shared pointer to `react::CallInvoker` and so we should make sure they both can deallocate when the JS engine wants it.

Well, if you do `global.ExpoModules = null` then you actually opt into some unexpected behavior 🤪
We can also prevent this case with such JS code:

```javascript
const ExpoModules = global.ExpoModules;
delete global.ExpoModules;
// And now use `ExpoModules` instead of `global.ExpoModules`.
```
This shouldn't deallocate the object 🤔 I can check out how that works in #15848